Libs/MRML/Logic/vtkMRMLColorLogic.cxx

```diff
@@ -1076,3 +1094,699 @@ vtkMRMLProceduralColorNode* vtkMRMLColorLogic::CopyProceduralNode(vtkMRMLColorNo
+      StandardTerm category, type, modifier;
+      this->ParseTerm(tokens[2],category);
+      this->ParseTerm(tokens[3],type);
+      this->ParseTerm(tokens[4],modifier);
+
+      // for now region doesn't appear in the file
+      StandardTerm region, regionModifier;
+      addFlag = addFlag && this->AddTermToTerminologyMapping(lutName, labelValue, category, type, modifier, region, regionModifier);
+     }
+  }
+  std::cout << this->colorCategorizationMaps[lutName].size()
+            << " terms were read for Slicer LUT " << lutName << std::endl;
+
+  return status && addFlag && assocFlag;
+}
+
+//-------------------------------------------------------------------------------
+bool vtkMRMLColorLogic::
+  LookupCategorizationFromLabel(int label, ColorLabelCategorization& labelCat, const char *lutName)
+{
+  bool success = false;
+
+
+  if (this->TerminologyExists(lutName))
+    {
+    // set the label value so that if it's not found, it's still a valid categorisation
+    labelCat.LabelValue = label;
+    if (this->colorCategorizationMaps[lutName].find(label) !=
+      this->colorCategorizationMaps[lutName].end())
+      {
+      labelCat = this->colorCategorizationMaps[lutName][label];
+      success = true;
+      }
+    }
+  return success;
+}
+
+//---------------------------------------------------------------------------
+std::string vtkMRMLColorLogic::
+  GetTerminologyFromLabel(std::string categorization, std::string standardTerm, int label, const char *lutName)
+{
+  std::string returnString;
+  ColorLabelCategorization labelCat;
+  if (this->LookupCategorizationFromLabel(label, labelCat, lutName))
+    {
+    StandardTerm term;
+    if (categorization.compare("SegmentedPropertyCategory") == 0)
+      {
+      term = labelCat.SegmentedPropertyCategory;
+      }
+    else if (categorization.compare("SegmentedPropertyType") == 0)
+      {
+      term = labelCat.SegmentedPropertyType;
+      }
+    else if (categorization.compare("SegmentedPropertyTypeModifier") == 0)
+      {
+      term = labelCat.SegmentedPropertyTypeModifier;
+      }
+    else if (categorization.compare("AnatomicRegion") == 0)
+      {
+      term = labelCat.AnatomicRegion;
+      }
+    else if (categorization.compare("AnatomicRegionModifier") == 0)
+      {
+      term = labelCat.AnatomicRegionModifier;
+      }
+    // now get the requested coding from the standard term
+    if (standardTerm.compare("CodeValue") == 0)
+      {
+      returnString = term.CodeValue;
+      }
+    else if (standardTerm.compare("CodeMeaning") == 0)
+      {
+      returnString = term.CodeMeaning;
+      }
+    else if (standardTerm.compare("CodingSchemeDesignator") == 0)
+      {
+      returnString = term.CodingSchemeDesignator;
+      }
+    }
+  return returnString;
+}
+
+//---------------------------------------------------------------------------
+bool vtkMRMLColorLogic::
+LookupLabelFromCategorization(ColorLabelCategorization& labelCat, int& label, const char *lutName)
```

Comment:
I am not sure we need this function. It was present in Reporting logic since we did not create color tables on the fly. Currently it is used only by the test.

As a side-note, we should probably mention in the commit that some of the methods introduced here were migrated from Reporting logic.